app/code/Magento/Sales/Block/Adminhtml/Order/View/Tab/History.php

```diff
@@ -303,10 +348,39 @@ public static function sortHistoryByTimestamp($a, $b)
     {
         $createdAtA = $a['created_at'];
         $createdAtB = $b['created_at'];
+        
+        if ($createdAtA->getTimestamp() === $createdAtB->getTimestamp()) {
+            return $a['entity_id'] <=> $b['entity_id'];
+        }
 
         return $createdAtA->getTimestamp() <=> $createdAtB->getTimestamp();
     }
 
+    /**
+     * Comparison For Sorting History By Timestamp and entity_id
+     *
+     * @param mixed $a
+     * @param mixed $b
+     * @return int
+     */    
+    private function sortHistory($a, $b)
+    {
+        $createdAtA = $a['created_at'];
+        $createdAtB = $b['created_at'];
+        
+        if ($createdAtA->getTimestamp() === $createdAtB->getTimestamp()) {
+            if ($a['type'] < $b['type']) {
+                return 1;
+            } elseif ($a['type'] > $b['type']) {
+                return -1;
+            } else {
+                return $a['entity_id'] <=> $b['entity_id'];
+            }
+        }
+
+        return $createdAtA->getTimestamp() <=> $createdAtB->getTimestamp();
```

Comment:
I would suggest something like this:
```suggestion
        $result = $this->sortHistoryByTimestamp($a, $b);
        if (0 !== $result) {
            return $result;
        }
        
        $result = $a['type'] <=> $b['type'];
        if (0 !== $result) {
            return $result;
        }
        
        return $a['entity_id'] <=> $b['entity_id'];
```